### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+exclude_paths:
+  - docs
+  - example_postman
+  - examples
+  # skip the test directories because it the spec blocks aggressively show up as
+  # duplicates
+  - packages/*/test
+  - src
+  - test
```
Comment (Josh-Dykstra) [inline .codeclimate.yml]: should we add resports here?

File: Gruntfile.circle.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    eslint: {
+      options: {
+        format: process.env.XUNIT ? `junit` : `stylish`,
+        outputFile: process.env.XUNIT && `<%= xunitDir %>/eslint.xml`
+      },
+      all: [
+        `./packages/*/src/**/*.js`,
+        `./packages/*/test/**/*.js`,
+        `./packages/*/*.js`
```
Comment (Josh-Dykstra) [inline Gruntfile.circle.js]: Any other folders we should lint? What about the examples?

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    XUNIT: true
+  node:
+    version: 4
+
+dependencies:
+  cache_directories:
+    - node_modules
+    - ../spark-js-sdk--test-users
+    # TODO deploy script should update this list automatically
+    # circleci doesn't appear to support cache wildcards
```
Comment (Josh-Dykstra) [inline circle.yml]: I don't see it in the docs either :/

File: .codeclimate.yml (hunk 1 above)
Comment (ianwremmel) [inline .codeclimate.yml]: already excluded via .gitignore

File: run.sh [hunk 4]
```diff
@@ -52,5 +56,17 @@ else
...
 fi
 
+# Make sure we have the github.com remote setup
+set +e
+git remote | grep -qc ghc
+IS_MISSING_GHC_REMOTE=$?
+set -e
+if [ $IS_MISSING_GHC_REMOTE -eq "1" ]; then
+  git remote add ghc git@github.com:ciscospark/spark-js-sdk.git
+fi
```
Comment (Josh-Dykstra) [inline run.sh]: Great!

File: tooling/circle-cli/lib/common-options.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+module.exports = {
+  auth: {
```
Comment (Josh-Dykstra) [inline tooling/circle-cli/lib/common-options.js]: No shortcut for auth? :)

**Pull request merged**